secrets/secrets-kubernetes/pkg/secrets/ed25519.go

```diff
@@ -0,0 +1,64 @@
+package secrets
+
+import (
+	"errors"
+	"fmt"
+
+	jwt "github.com/golang-jwt/jwt/v5"
+	"github.com/nats-io/nkeys"
+)
+
+var ErrEd25519Verification = errors.New("ed25519: verification error")
+
+// SigningMethodNats implements the Ed25519 family.
+type SigningMethodNats struct{}
+
+// Specific instance for Ed25519 - nkeys edition
+var (
+	SigningMethodEd25519 *SigningMethodNats
+)
+
+func init() {
+	SigningMethodEd25519 = &SigningMethodNats{}
+	jwt.RegisterSigningMethod(SigningMethodEd25519.Alg(), func() jwt.SigningMethod {
+		return SigningMethodEd25519
+	})
+}
+
+func (m *SigningMethodNats) Alg() string {
+	return "Ed25519"
+}
+
+// Verify implements token verification for the SigningMethod.
+func (m *SigningMethodNats) Verify(signingString string, sig []byte, key interface{}) error {
```

Comment:
Forgive my Go newbie status... Shouldn't this function just take an `nkeys.KeyPair` instead of `interface{}`? What's the benefit to keeping it more generic here

Reply:
It's implementing an upstream interface for jwt validation https://github.com/golang-jwt/jwt/blob/main/signing_method.go#L14

you're right. feels like this interface was written pre-generics era.